pkg/installmanager/installmanager.go

```diff
@@ -864,6 +865,58 @@ func (m *InstallManager) generateAssets(cd *hivev1.ClusterDeployment, workerMach
 		}
 	}
 
+	// Day 0: Add MachinePool label to MachineSet manifests
+
+	mapPoolsByType := map[string]string{}
+	mpL := &hivev1.MachinePoolList{}
+	if err := m.DynamicClient.List(context.Background(), mpL, &client.ListOptions{Namespace: m.Namespace}); err != nil {
+		return err
+	}
+	for _, pool := range mpL.Items {
+		if pool.Spec.ClusterDeploymentRef.Name == cd.Name {
+			mapPoolsByType[pool.Spec.Name] = pool.Name
+		}
+	}
```

Comment:
Redundant `List` call. Let's collapse this chunk with the guts of `loadWorkerMachinePool()`. Make that func instead `loadMachinePools()` and produce `map[string]*MachinePool`; replace the `workerMachinePool` arg into this func with that map; then below this chunk you can do `workerMachinePool := mapPoolsByType["worker"]` to leave the rest working as it does today. `patchLabelMachineSetManifsts` will need a trivial refactor to extract the `pool.Name` from the map value.